### PR TITLE
parameterize password strength tests

### LINKED ISSUE #479: Make password strength tests parameterized
In `UserTest.java` , we have the following three tests:

```java
void testWeakPasswordIsInvalid() {
    // given
    User user = userWithoutPassword().password("123456789")
                                     .build();
    // when
    tryToSaveInvalidUser(user);
    // then
    assertThat(userRepository.count()).isEqualTo(initialNumberOfUsers);
}
@Test
void testFairPasswordIsInvalid() {
    // given
    User user = userWithoutPassword().password("aPassWorD")
                                     .build();
    // when
    tryToSaveInvalidUser(user);
    // then
    assertThat(userRepository.count()).isEqualTo(initialNumberOfUsers);
}
@Test
void testGoodPasswordIsInvalid() {
    // given
    User user = userWithoutPassword().password("testPa$$123")
                                     .build();
    // when
    tryToSaveInvalidUser(user);
    // then
    assertThat(userRepository.count()).isEqualTo(initialNumberOfUsers);
}
```

**Context:**

'Weak', 'fair' and 'good' refer to password strengths (you can try this out by running the app). The problem here is that the shape of each of these three tests are the same: we construct an invalid user, try to save it and then assert that the number of users saved (userRepository.count()) has not changed (== initialNumberOfUsers).

**What we want:**

We want to eliminate the code duplication. As you may have noticed, the only thing that's changing is the password. We can combine these three tests into one and pass in as input strings that correspond to a weak, fair and good password. The way we do this in JUnit is by creating a parameterised test

### PR DESCRIPTION
**Pull request by SeonghoLee4**
## Summary of change

Added a new Junit Test case (eliminate duplicate tests - testPasswordIsInvalid)

## Related issue

Closes #479 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

### CODE REVIEW

File: src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java [hunk 1]
```diff
@@ -61,7 +63,21 @@ void testValidUserSaved() {
...
+    void testPasswordIsInvalid(String input) {
+            // given
+            User user = userWithoutPassword().password(input)
+                                             .build();
+
+            // when
+            tryToSaveInvalidUser(user);
 
+            // then
+            assertEquals(initialNumberOfUsers, userRepository.count());
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java:78]: Could you please change this to use an [AssertJ ](https://assertj.github.io/doc/)assertion (in line with our contributing guidelines)?

File: src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java [hunk 2]
```diff
@@ -61,7 +63,21 @@ void testValidUserSaved() {
                    .username("username")
                    .email("email");
     }
+    
+    @ParameterizedTest
+    @ValueSource(strings = {"123456789", "aPassWorD", "testPa$$123"})
+    void testPasswordIsInvalid(String input) {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java:69]: Your fork is quite far behind master. In the latest `UserTest` version on master, we're following a format where the display name is [class to test] 'should'. In this test class, the class has the display name `User should`. Each method then follows on from that sentence to complete a sentence (e.g. `notHaveAWeakPassword`). 

In line with our style guide, could you please rename your method to follow this format?

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for working on this! I've left some feedback for you to address.

From looking at the Travis log, it seems your error is because there are no `assertEquals` methods (overrides) that takes two `Long`s. However, we're using AssertJ assertions.

As this branch on your fork is far behind master on this repository, I'd recommend:
1. Creating a remote to point towards [upstream](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork) (this repository)
2. Then, [sync](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork) your branch by fetching `master` and then rebasing (not merging)

Once you've brought your branch level with master, you should be able to see locally how we're naming our test methods.

File: src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java (hunk 1 above)
Comment (SeonghoLee4) [reply src/test/java/com/karankumar/bookproject/backend/entity/account/UserTest.java:78]: I will check it out !
Thanks for the feedback.

**Comment by knjk04** [comment]
@SeonghoLee4 How are you getting on with addressing the review feedback?

**Comment by SeonghoLee4** [comment]
> @SeonghoLee4 How are you getting on with addressing the review feedback?

I haven't had enough time to work on it. I will get to working on it pretty soon!

**Comment by knjk04** [comment]
@SeonghoLee4 Thanks. I also noticed just now that your branch does name does not contain the issue number. Could you please rename the branch (e.g. using `git branch -m <new-name>`. You'll also need to open a new PR, so I'll close this one.

Feel free to reach out to us on Slack if you need a hand

**Comment by SeonghoLee4** [comment]
@knjk04 checked!

**Pull request closed without merging**